FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp

```diff
@@ -101,28 +101,42 @@ DEF_OP(LoadNamedVectorConstant) {
     }
   }
   // Load the pointer.
-  ldr(TMP1, STATE_PTR(CpuStateFrame, Pointers.Common.NamedVectorConstantPointers[Op->Constant]));
+  auto GenerateMemOperand = [this](uint8_t OpSize, uint32_t NamedConstant, FEXCore::ARMEmitter::Register Base) {
+    const auto ConstantOffset = offsetof(FEXCore::Core::CpuStateFrame, Pointers.Common.NamedVectorConstants[NamedConstant]);
 
+    if (ConstantOffset <= 255 || // Unscaled 9-bit signed
+        ((ConstantOffset & (OpSize - 1)) == 0 &&  ConstantOffset / OpSize <= 4095)) /* 12-bit unsigned scaled */ {
```

Comment:
Integer division is potentially expensive, and I don’t expect the compiler knows that op size is a power of two and hence support a more efficient ctz + shift implementation. Do we have a relevant utility function for dividing by powers of 2? If not, perhaps, now is the time.

Reply:
Added a log2 divider helper and unittest for it.